Feature 488: Make all download data downloadable

Stoyan0v opened the pull request:
Fixes #488 

Currently when you hit the download button on "Download your data" page in some cases you are redirected to the source, where you can copy your data. 
In this PR this issue is addressed and now all buttons will download a file, instead opening the content.

The only thing that I am not sure here is the content type. Currently it's hadrcoded to `application/octet-stream`, but I don't think that we will have different types for now. We can easy address this by using `get_mime_type` or `file_mime_type`

markkap commented:
@Stoyan0v I fixed some possible PHP notice with cleaning the output buffer, and changed to safe remote get so the VIP linter will be quiet.

Two other things
1. Do you think it will be better to have a date as part of the file name, and maybe use wordlift-dataset.xxx pattern.
2. Lets remove all the silencing of linter warning except for the body one. Just validate that the out parameter is one of the allowed 4, and give specific error when it is not? 
3. Probably the admin is the one doing it, so asking him to contact the admin if something fails is funny :) need to think of some better text maybe just "communication problem"?

Stoyan0v commented:
Hey @markkap . Thanks you for the feedback. I will appreciate if you can take look again :)

> Do you think it will be better to have a date as part of the file name, and maybe use wordlift-dataset.xxx pattern.

Make sense. I've added the timestamp to filename.

> Lets remove all the silencing of linter warning except for the body one. Just validate that the out parameter is one of the allowed 4, and give specific error when it is not?

Actually the response types are only two. I've validate the response content type and if it's not valid a "good" error message is being displayed.

>Probably the admin is the one doing it, so asking him to contact the admin if something fails is funny :) need to think of some better text maybe just "communication problem"?

You are right. I've changed the error messages.
